### PR TITLE
47 statistical parameters improvement

### PR DESCRIPTION
**Pull request by Dawid64**
Implemented meta features from paper: "Initializing bayesian hyperparameter optimization via meta-learning".
Additionally fixed preprocessing issue.
Added tests for each dataset parameter extractors.

### CODE REVIEW

File: turbo_ml/turboml.py
Comment (Tole-k) [inline turbo_ml/turboml.py:1]: when passing parameters to hpo tuner accesing via . doesnt work anymore and should be changed to [''] just like in turboml_experimental

File: turbo_ml/meta_learning/meta_model/meta_model_search.py
Comment (Tole-k) [inline turbo_ml/meta_learning/meta_model/meta_model_search.py:1]: You seem to blacklist NeuralNetworkModel, what's the reason behind it? is there some problem with it? (apart from optuna's inablity to tune it thoroughly enough, which would actually be a valid reason)

**Review by Tole-k — CHANGES_REQUESTED** [review]

File: turbo_ml/meta_learning/meta_model/meta_model_search.py
Comment (Dawid64) [reply turbo_ml/meta_learning/meta_model/meta_model_search.py:1]: This is due to inability of NNModel to autocomplete it's input, as it was the only model that needed additional user input and made could easily make problems that could not always be seen so easily (Kind of issue that can occur once per 100 runs which is the worst kind of issue)

File: turbo_ml/turboml.py
Comment (Dawid64) [reply turbo_ml/turboml.py:1]: Should be solved in new commits

File: turbo_ml/meta_learning/dataset_parameters/sota.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            return results_dict
+        return np.concatenate(results)
+
+
+class SimpleMetaFeatures(MetaFeature):
+    def __call__(self, dataset: pd.DataFrame, target_data: pd.Series, as_dict: bool = False) -> np.ndarray | dict:
+        types = dataset.dtypes
+        num_features = len(dataset.dtypes)
+        log_num_cols = np.log2(num_features)
+        num_patters = len(dataset)
```
Comment (adamtms) [inline turbo_ml/meta_learning/dataset_parameters/sota.py:65]: Typo. num_patters instead of num_patterns

**Review by Tole-k — APPROVED** [review]

**Pull request merged**